pkg/daemon/update.go

```diff
@@ -2684,6 +2690,30 @@ func runCmdSync(cmdName string, args ...string) error {
 	return nil
 }
 
+var (
+	podmanSigstoreSupported      sync.Once
+	podmanSigstoreSupportedValue bool
+)
+
+func podmanSupportsSigstore() bool {
+	podmanSigstoreSupported.Do(func() {
+		// https://issues.redhat.com/browse/OCPBUGS-38809 failed for base image 4.11 or older, OCP 4.12 is with podman 4.4.1
+		// returns false if podman version is less than 4.4.1
+		cmd := exec.Command("podman", "version", "-f", "{{.APIVersion}}")
+		out, err := cmd.CombinedOutput()
+		if err != nil {
+			klog.Errorf("failed to run podman version: %v", err)
+			podmanSigstoreSupportedValue = false
+			return
+		}
+		sigstorePodman := "4.4.1"
+		// Example version format: 5.3.0-rc1
+		imgPodmanVersion := strings.TrimSpace(string(out))
+		podmanSigstoreSupportedValue = semver.New(imgPodmanVersion).Compare(*semver.New(sigstorePodman)) >= 0
```

Comment:
so this will panic if it's invalid for some reason https://github.com/coreos/go-semver/blob/main/semver/semver.go#L65
can you use [NewVersion](https://github.com/coreos/go-semver/blob/main/semver/semver.go#L52) and gracefully handle the error?